### PR TITLE
Add Charcoal

### PR DESCRIPTION
**Pull request by Extex101**
Adds charcoals to Minetest Game
default_hardwood_charcoal
Charcoal. The result of cooking trees
Required for #2472

### CODE REVIEW

**Comment by paramat** [comment]
So the primary benefit is crafting torches without needing coal? Interesting and seems useful.

**Comment by paramat** [comment]
Ah it's for gunpowder crafting too.
In that case, i would like core dev support for #2472 to give charcoal more usefulness.

**Comment by Df458** [comment]
I think the idea of charcoal is a good one and I support it, but I question the need for two types of charcoal. Even if it's more 'true to life', the end result is that it takes twice as much storage space and you can only stack one of the two in a furnace at a time.

There's no benefit for decoration, and the one gameplay difference I see here is that one of the two types can't craft torches. That actually kinda sucks imo, it means that spawning in a cold biome will put you at even more of a disadvantage than it already does. Warmer biomes provide apples, blueberries, cotton, wheat and now free torches. Colder biomes offer... ice? Snow? Nothing useful that I can think of. I don't want to hijack this thread too much with that line of thinking, but I believe that making somewhat unbalanced biomes *even less balanced* is something that should be avoided, especially when there's no apparent benefit to doing so.

**Comment by paramat** [comment]
I think i also prefer only 1 charcoal type, to keep it simpler, then we don't need the extra group.

**Comment by Extex101** [comment]
Done!

File: mods/default/README.txt [hunk 1]
```diff
@@ -248,7 +248,7 @@ Topywo (CC BY-SA 3.0)
 
 Extex101 (CC BY-SA 3.0)
   default_large_cactus_seedling.png
-
+  default_hardwood_charcoal.png
```
Comment (mbartlett21) [inline mods/default/README.txt]: This should be 
```suggestion
  default_charcoal.png
```

File: mods/default/crafting.lua [hunk 2]
```diff
@@ -89,6 +89,14 @@ minetest.register_craft({
 	}
 })
 
+minetest.register_craft({
+	output = "default:torch 3",
+	recipe = {
+		{"default:hardwood_charcoal"},
```
Comment (DS-Minetest) [inline mods/default/crafting.lua]:
```suggestion
		{"default:charcoal"},
```

**Comment by DS-Minetest** [comment]
Ew. such a minecrafty way of making charcoal.
There are mods that add much nicer ways to gain charcoal, like [`another_charcoal`](https://forum.minetest.net/viewtopic.php?t=13549) or [`techage`](https://github.com/joe7575/techage).

**Comment by paramat** [comment]
Cooking in a furnace is ok for me, this needs to be simple.
Because charcoal takes time to create i think the cooktime should be quite long, such that charcoal is fairly expensive to produce. Looks like the default cooktime is 3 so 10 is the right idea.

**Comment by Extex101** [comment]
So It's just In the awaiting approval stage now?

**Comment by paramat** [comment]
I'll review more carefully later, we're busy with releasing 5.1.0.

**Comment by paramat** [comment]
I would like to have core dev support for #2472 first to give charcoal more uses.

**Comment by paramat** [comment]
I think the proposed improvements to TNT/gunpowder crafting could just use nitre, not charcoal, for simplicity.
I think this PR is justified by how it makes torches possible without coal.

**Comment by Extex101** [comment]
Ok but gunpowder should at least have coal if not charcoal.
Gunpowder wouldn't possibly work without something to burn

**Comment by paramat** [comment]
Yes, coal for gunpowder is fine for me.

**Comment by MoNTE48** [comment]
@Extex101 rebase needed! 😉

**Comment by paramat** [comment]
This somewhat depends on whether #2472 goes ahead, so no need to rebase yet.

**Comment by paramat** [comment]
#2472 will probably not go ahead, so the usage for TNT crafting is no longer relevant.
But still good for crafting torches without needing coal.

Any core dev concept support?

File: mods/default/craftitems.lua [hunk 3]
```diff
@@ -350,3 +350,9 @@ minetest.register_craftitem("default:blueberries", {
 	groups = {food_blueberries = 1, food_berry = 1},
 	on_use = minetest.item_eat(2),
 })
+
+minetest.register_craftitem("default:charcoal", {
+	description = S("Charcoal"),
+	inventory_image = "default_charcoal.png",
+	groups = {coal = 1, flammable = 1}
```
Comment (paramat) [inline mods/default/craftitems.lua:357]: Why the 'coal' group?

**Comment by paramat** [comment]
Feature with no core dev concept support, so closing.

**Pull request closed without merging**